buildkite/resource_agent_token.go

```diff
@@ -45,35 +46,88 @@ func (at *AgentTokenResource) Configure(ctx context.Context, req resource.Config
 
 func (at *AgentTokenResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	var plan, state AgentTokenStateModel
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
 
-	apiResponse, err := createAgentToken(ctx,
-		at.client.genqlient,
-		at.client.organizationId,
-		plan.Description.ValueStringPointer(),
-	)
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
 
-	if err != nil {
-		resp.Diagnostics.AddError(err.Error(), err.Error())
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	timeout, diags := at.client.timeouts.Create(ctx, DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
-	state.Description = types.StringPointerValue(apiResponse.AgentTokenCreate.AgentTokenEdge.Node.Description)
-	state.Id = types.StringValue(apiResponse.AgentTokenCreate.AgentTokenEdge.Node.Id)
-	state.Token = types.StringValue(apiResponse.AgentTokenCreate.TokenValue)
-	state.Uuid = types.StringValue(apiResponse.AgentTokenCreate.AgentTokenEdge.Node.Uuid)
+	var r *createAgentTokenResponse
+	retry.RetryContext(ctx, timeout, func() *retry.RetryError {
+		var err error
+		r, err = createAgentToken(ctx,
+			at.client.genqlient,
+			at.client.organizationId,
+			plan.Description.ValueStringPointer(),
+		)
+
+		if err != nil {
+			if isRetryableError(err) {
+				return retry.RetryableError(err)
+			}
+			resp.Diagnostics.AddError(
+				"Unable to create agent token",
+				fmt.Sprintf("Unable to create agent token: %s", err.Error()),
+			)
+			return retry.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
```

Comment:
If `retry.RetryContext` returns an error the Create function will still proceed to set the state. Any errors should be blocked a this point.